grakn-graql/src/main/java/ai/grakn/graql/internal/gremlin/sets/ResourceIndexFragmentSet.java

```diff
@@ -96,7 +97,10 @@ private static void optimise(
 
     private static Stream<ValueFragmentSet> equalsValueFragments(Collection<EquivalentFragmentSet> fragmentSets) {
         return fragmentSetOfType(ValueFragmentSet.class, fragmentSets)
-                .filter(valueFragmentSet -> valueFragmentSet.predicate().equalsValue().isPresent());
```

Comment:
I won't pretend that I understand what is happening here, but the tests cover the issue I had so I am happy based on that.